controllers/users.js

```diff
@@ -0,0 +1,41 @@
+const express = require("express")
+const router = express.Router()
+const bcrypt = require("bcrypt")
+const User = require("../models/user")
+const jwt = require("jsonwebtoken")
+
+const SALT_LENGTH = 12
+
+router.post("/signup", async (req, res) => {
+	try {
+		const userInDatabase = await User.findOne({ username: req.body.username })
+
+		if (userInDatabase) {
+			return res.json({
+				error:
+					"That username is already taken. How about trying a different one?",
+			})
+        }
+        
+		if (req.body.password !== req.body.confirmPassword) {
+			return res.json({
+				error:
+					"Oops! It looks like your passwords don’t match. Please double-check and try again!",
+			})
+		}
```

Comment:
Not sure if we'll need to do this. I think most likely we'll verify matching passwords on the front end before passing it to the service. If that's the case, there won't be a confirm password field by the time it reaches the back end.

Reply:
Happy to remove it once the front end code is completed !